### ISSUE TITLE
QML code doesn't seem to use provided datapath, fails location system-installed code

### ISSUE THREAD

**Issue by akien-mga** [Andrettin/Wyrmgus#157]
- Version: 4.0.5
- OS: Mageia 8

Hi Andrettin :)

I just updated my Mageia package to 4.0.5 (from 3.5.4), and I now get this issue when starting the game using the launch script:
```
$ pwd
/home/akien
$ wyrmsun 
QQmlApplicationEngine failed to load component
file:///home/akien/interface/Main.qml: No such file or directory
Invalid language_family instance: "celtic".
$ wyrmgus -d /usr/share/games/wyrmsun/
QQmlApplicationEngine failed to load component
file:///home/akien/interface/Main.qml: No such file or directory
Invalid language_family instance: "celtic".
```

The `interface` folder also doesn't seem installed by CMake:
```
$ ls /usr/share/games/wyrmsun/
data/  graphics/  maps/  music/  oaml.defs  scripts/  sounds/  translations/
```

## Steps to reproduce

In Wyrmgus dir:
```
cmake -DCMAKE_INSTALL_PREFIX=/usr -DENABLE_USEGAMEDIR=OFF
make
make install
```

In Wyrmsun dir:
```
cmake -DBINDIR=games -DDATADIR=share/games/wyrmsun
make install
```

**Comment by Andrettin** [comment]
Thanks for the bug report, Akien! I added the "interface" folder to the folder installation list with the following commit:
https://github.com/Andrettin/Wyrmsun/commit/60963de524521b1d845a08436ddfc4cdef7ab026

Otherwise, I'm not sure what is causing this issue... ENABLE_USEGAMEDIR relates only to logs, save games and the like, not to the game's data, so if it found it before, it should be finding it now I suppose.

Though note that std::filesystem::current_path() is now used for loading data which is in a new format. This would not affect the QML loading, but could be reason why the "celtic" language family definition wasn't present.

**Comment by akien-mga** [comment]
Note that `ENABLE_USEGAMEDIR` is not what's causing the issue, I just documented the build options I use.

Stratagus relies on the `-d` argument to locate game data which can be installed in any arbitrary game path (the engine binary itself doesn't know where Wyrmsun data is installed), so my assumption is that the QML loading code doesn't respect this logic, and instead expects a specific repo setup (typically what you use for static distribution).

**Comment by Andrettin** [comment]
@akien-mga You are absolutely right! I found out what the issue was, and implemented fixes for it:
https://github.com/Andrettin/Wyrmgus/commit/b876938da35d821588f243aff389cd46fe422597
https://github.com/Andrettin/Wyrmgus/commit/0b24276c8780d5f309d43758edc7318393ff0ca6
https://github.com/Andrettin/Wyrmgus/commit/f8a045c54a6afead46bc9dc1daa92db0e33782fc

With these changes, I tested the "-d" command line option under Windows, and it worked for me.

**Comment by akien-mga** [comment]
Seems fixed in 4.1.2, thanks!
